arch/risc-v/src/common/riscv_percpu.h

```diff
@@ -38,10 +38,11 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#ifdef __ASSEMBLY__
-#define SCRATCH_HARTID_OFFSET   (0 * INT_REG_SIZE)
-#else
-#define SCRATCH_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
+#define RISCV_PERCPU_HARTID_OFFSET   (0 * INT_REG_SIZE)
+
+#ifndef __ASSEMBLY__
+static_assert(RISCV_PERCPU_HARTID_OFFSET == offsetof(riscv_percpu_s, hartid),
```

Comment:
I think we need to move static assert to C file

Reply:
Done.